src/services/UserService.ts

```diff
@@ -0,0 +1,42 @@
+import { User } from '../entities/users';
+import { MysqlDataSource } from '../database/datasource';
+import { throwError } from '../utils/ErrorResponse';
+import { Repository } from 'typeorm';
+
+export class UserService {
+  private userRepository: Repository<User>
+  constructor() {
+    this.userRepository = MysqlDataSource.getRepository(User);
```

Comment:
Use `Dependency Injection`

Reply:
Implemented `dependency-injection` using `tsyring`. Thought to implement basic `constructor based dependency-injection` but gave this a try.